### PR TITLE
Part 8 integration specs and fix n+1

### PR DESCRIPTION
**Pull request by henokkhm**
This pull request addresses two main issues:
   1. fix the *N+1 problem* when fetching all posts and comments of a user
   2. use **Capybara** to write integration tests for each view 

The screenshots of the app's logs before and after implementing the fix for the *N+1 problem* are shown below.

### Before
blog-app-n+1-before

### After
blog-app-n+1-after

## Summary
- Fix n+1 query for post comments
- Update database seeds script
- Use db seed for all tests
- Write test cases for each of the following views
  - posts index
  - post show
  - users index
  - user show
- **Remove outdated tests that check for placeholder text**

### All tests are passing
Screenshot from 2023-11-09 21-35-26

### CODE REVIEW

File: spec/requests/integration/posts/post_index_spec.rb
Comment (fobadara) [inline spec/requests/integration/posts/post_index_spec.rb:1]: [REQUIRED] Kindly include tests that asserts you can see a section for pagination if there are more posts than fit on the view.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors 🟢 

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Your tests are failing as you can see below. Please try to make them pass 🙏 
image


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by henokkhm** [comment]
Hi @codepantha, 

All the tests in the last commit are passing on my machine.
image

Please make sure that you run `bin/rails db:environment:set RAILS_ENV=development` and then `bin/rails db:reset` to populate your database with the records in **db/seeds.rb**

You can open the **/users** route to double-check that the data that the tests depend on was successfully added.

image

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @henokkhm         👍 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Rails features  ✔️
- [x] Kudus on fixing all changes suggested by the last reviewer   
- [x] Kudus on meeting the requirements   👏 

### Confirmed view:
image


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**